Content.Shared/Payload/Components/PayloadTriggerComponent.cs

```diff
@@ -0,0 +1,44 @@
+using Robust.Shared.Prototypes;
+
+namespace Content.Shared.Payload.Components;
+
+/// <summary>
+///     Component for providing the means of triggering an explosive payload. Used in grenade construction.
+/// </summary>
+/// <remarks>
+///     This component performs two functions. Firstly, it will add or remove other components to some entity when this
+///     item is installed inside of it. This is intended for use with constructible grenades. For example, this allows
+///     you to add things like <see cref="OnUseTimerTriggerComponent"/>, or <see cref="TriggerOnProximityComponent"/>.
+///     This is required because otherwise you would have to forward arbitrary interaction directed at the casing
+///     through to the trigger, which would be quite complicated. Also proximity triggers don't really work inside of
+///     containers.
+///
+///     Secondly, if the entity that this component is attached to is ever triggered directly (e.g., via a device
+///     network message), the trigger will be forwarded to the device that this entity is installed in (if any).
+/// </remarks>
+[RegisterComponent]
+public sealed class PayloadTriggerComponent : Component
```

Comment:
if this is in shared should it be networked?

Reply:
I mostly just had it in shared for the sake of entity-whitelists.  The only reason it might need to be networked is if the trigger component is ever dynamically added to entities? I guess there's no harm in putting a networked attribute on it, but there is no need for state handling/sending.